src/content/en/updates/2018/03/clipboardapi.md

```diff
@@ -0,0 +1,269 @@
+pasted.  Imagine a web page that silently copies `rm -rf /` or a [decompression
+bomb image] to your clipboard.
+
+Giving web pages unfettered read access to the clipboard is even more
+troublesome. Users routinely copy sensitive information like passwords and
+personal details to the clipboard, which could then be read by any page without
+them ever knowing.
+
+As with many new APIs, [`navigator.clipboard`] is only supported for pages
+served over HTTPS. To help prevent abuse, clipboard access is only allowed when
+a page is the active tab. Pages in active tabs can write to the clipboard
+without requesting permission, but reading from the clipboard always requires
+permission.
+
+To make things easier, two new permissions for copy & paste can be used with the
+[Permissions API]. There are separate permissions for read (paste) and write
+(copy):
+
+    { name: 'clipboard-read' }
+    { name: 'clipboard-write' }
+
+<img src="/web/updates/images/2018/03/clipboardapi/prompt.png" alt="Screenshot
+of the permissions prompt shown when attempting to read from the clipboard.">
+
+
+As with anything using the Permissions API, it's possible to check if your app
+has permission to interact with the clipboard:
+
+    navigator.permissions.query({
+      name: 'clipboard-read',
+      allowWithoutGesture: false
+    }).then( permissionStatus => {
+      // Will be 'granted', 'denied' or 'prompt':
+      console.log(permissionStatus.state);
+
+      // Listen for changes to the permission state
+      permissionStatus.onchange = () => {
+        console.log(permissionStatus.state);
+      };
+    });
+
+Here's where the "async" part of the Clipboard API really comes in handy though:
+attempting to read or write clipboard data will automatically prompt the user
+for permission if it hasn't already been granted.  Since the API is
+promise-based this is completely transparent, and a user denying clipboard
+permission rejects the promise so the page can respond appropriately.
+
+Since Chrome only allows clipboard access when a page is the current active tab,
+you'll find some of the examples here don't run quite right if pasted directly
+into DevTools, since DevTools itself is the active tab. There's a trick: we need
+to defer the clipboard access using setTimeout, then quickly click inside the
+page to focus it before the functions are called:
+
+    setTimeout( () => {
+      navigator.clipboard.writeText('Hello World!');
+    }, 2000);
+
+
+## Looking Back
+
+Prior to the introduction of the Async Clipboard API, we had a mix of different
+copy & paste implementations across web browsers.
+
+In most browsers, the browser's own copy and paste can be triggered using
+`document.execCommand('copy')` and `document.execCommand('paste')`.  If the text
+to be copied is a string not present in the DOM, we have to inject and select
+it:
+
+    button.addEventListener('click', e => {
+      let input = document.createElement('input');
```

Comment:
let -> const for these

Reply:
I think some of the update from the gdoc are missing? Can you double check this is the latest? 

I'll hold off on reviewing....

Reply:
I had exported to markdown prior to your review and was manually syncing. I'll go over the changes and make sure no others slipped through.

Reply:
K, I am fairly confident I've got all the changes sync'd up.